src/Renci.SshNet/Abstractions/DnsAbstraction.cs

```diff
@@ -36,72 +22,27 @@ public static IPAddress[] GetHostAddresses(string hostNameOrAddress)
-            var resolveCompleted = new ManualResetEvent(false);
-            NameResolutionResult nameResolutionResult = null;
-            DeviceNetworkInformation.ResolveHostNameAsync(new DnsEndPoint(hostNameOrAddress, 0), result =>
-            {
-                nameResolutionResult = result;
-                resolveCompleted.Set();
-            }, null);
-
-            // wait until address is resolved
-            resolveCompleted.WaitOne();
-
-            if (nameResolutionResult.NetworkErrorCode == NetworkError.Success)
-            {
-                var addresses = new List<IPAddress>(nameResolutionResult.IPEndPoints.Select(p => p.Address).Distinct());
-                return addresses.ToArray();
-            }
-            throw new SocketException((int)nameResolutionResult.NetworkErrorCode);
-#elif FEATURE_DATAGRAMSOCKET
-
-            // TODO we may need to only return those IP addresses that are supported on the current system
-            // TODO http://wojciechkulik.pl/csharp/winrt-how-to-detect-supported-ip-versions
-
-            var endpointPairs = DatagramSocket.GetEndpointPairsAsync(new HostName(hostNameOrAddress), "").GetAwaiter().GetResult();
-            var addresses = new List<IPAddress>();
-            foreach (var endpointPair in endpointPairs)
-            {
-                if (endpointPair.RemoteHostName.Type == HostNameType.Ipv4 || endpointPair.RemoteHostName.Type == HostNameType.Ipv6)
-                    addresses.Add(IPAddress.Parse(endpointPair.RemoteHostName.CanonicalName));
-            }
-            if (addresses.Count == 0)
-                throw new SocketException((int) System.Net.Sockets.SocketError.HostNotFound);
-            return addresses.ToArray();
-#else
-            throw new NotSupportedException("Resolving hostname to IP address is not implemented.");
-#endif // FEATURE_DEVICEINFORMATION_APM
-#endif
         }
 
         /// <summary>
         /// Returns the Internet Protocol (IP) addresses for the specified host.
         /// </summary>
         /// <param name="hostNameOrAddress">The host name or IP address to resolve.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used to signal the asynchronous operation should be canceled.</param>
         /// <returns>
         /// A task with result of an array of type <see cref="IPAddress"/> that holds the IP addresses for the host that
         /// is specified by the <paramref name="hostNameOrAddress"/> parameter.
         /// </returns>
         /// <exception cref="ArgumentNullException"><paramref name="hostNameOrAddress"/> is <see langword="null"/>.</exception>
         /// <exception cref="SocketException">An error is encountered when resolving <paramref name="hostNameOrAddress"/>.</exception>
-        public static Task<IPAddress[]> GetHostAddressesAsync(string hostNameOrAddress)
+        public static Task<IPAddress[]> GetHostAddressesAsync(string hostNameOrAddress, CancellationToken cancellationToken)
```

Comment:
I have a problem with this line. The Cancellation token is used only for .NET 6 +

Reply:
Can we just delete the whole file? It is not providing any abstraction now that the rest of the code is gone. `GetHostAddressesAsync` is only used in one place so we can move the #ifdef on the cancellation token to the callsite

Reply:
I don't think that TODO is worthwhile, and the library already makes liberal use of #ifdefs. Deleting the file sounds good to me